bpf/bpf_sock.c

```diff
@@ -515,18 +534,22 @@ sock6_skip_xlate(struct lb6_service *svc, union v6addr *address)
 		info = ipcache_lookup6(&IPCACHE_MAP, address,
 				       V6_CACHE_KEY_LEN);
 		if (info == NULL ||
-		     (svc->local_scope && info->sec_label != HOST_ID) ||
-		    (!svc->local_scope && info->sec_label != HOST_ID &&
-					  info->sec_label != REMOTE_NODE_ID))
+		    (svc->local_scope && info->sec_label != HOST_ID))
 			return true;
+		if (lb6_svc_is_external_ip(svc)) {
```

Comment:
Only thing that wasn't clear to me on this PR was the conversion of `(!svc->local_scope)` -> `lbX_svc_is_external_ip(svc)` but I take it that these are equivalent.

Reply:
It is in this case since the if condition in `sock{4,6}_skip_xlate()` is around `if (svc->local_scope || lb4_svc_is_external_ip(svc)) {`.